### DISCUSSION TITLE
typing '_' as part of a name results in no completion candidates

### DISCUSSION THREAD

**Discussion by ville-h** [company-mode/company-mode#1448] [Q&A] [answered]
Whilst programming some c++ typing '_' as part of a name results in no completion candidates being offered. I don't have the understanding to know whether none of the backsends(?) return anything or if it's company not showing anything.

My `company-backends` is `((company-capf company-yasnippet))`. My `completion-styles` is `(basic initials partial-completion substring)`. I am using eglot and yasnippet. Here is a simple test case to demonstrate:

```
auto abc_012() -> void {
}


auto main() -> int {
    // 2024-01-25 note: type "abc" on the line below and company will offer "abc_012()". continue typing
    // '_' and for some reason no completions are offered. continue typing '0' and "abc_012()" is offered
    // again
    
    return 0;
}
```

How could I improve my company, eglot or whatever configuration to have "abc_" result in "abc_012()" offered as a completion?

**Comment by dgutov** [comment, marked as answer]
Hi! That seems to be the fallout from the recent yasnippet-related changes.

I've pushed a fix to master, please test when the next build is published.

**Reply by ville-h** [reply]
Got the update just now and the issue seems to have been fixed. Thanks.
